ksql-engine/src/main/java/io/confluent/ksql/function/InternalFunctionRegistry.java

```diff
@@ -300,6 +309,45 @@ private void addJsonFunctions() {
         "ARRAYCONTAINS", ArrayContainsKudf.class));
   }
 
+  private void addUrlFunctions() {
+    KsqlFunction urlEncodeParam = new KsqlFunction(Schema.STRING_SCHEMA,
+        Arrays.asList(Schema.STRING_SCHEMA), "url_encode_param", UrlEncodeParamKudf.class);
+    addFunction(urlEncodeParam);
+
+    KsqlFunction urlDecodeParam = new KsqlFunction(Schema.STRING_SCHEMA,
```

Comment:
We need to use OPTIONAL (e.g. Schema.OPTIONAL_STRING_SCHEMA) schemas for the return values everywhere null can be returned. Otherwise I think the serializer will complain if we return a null value.

Reply:
@rodesai can you please explain how this relates to the serializer? What serializer?

Reply:
I mean the serializer thats invoked before the records are written to Kafka. Those use connect converters now, which will validate the values being serialized against the schema.